olive/systems/local.py

```diff
@@ -22,28 +22,31 @@ def __init__(self, accelerators: List[str] = None):
     def run_pass(
         self,
         the_pass: Pass,
-        model: OliveModel,
+        model_config: ModelConfig,
         data_root: str,
         output_model_path: str,
         point: Optional[Dict[str, Any]] = None,
-    ) -> OliveModel:
+    ) -> ModelConfig:
         """
         Run the pass on the model at a specific point in the search space.
         """
-        return the_pass.run(model, data_root, output_model_path, point)
+        model = model_config.create_model()
+        output_model = the_pass.run(model, data_root, output_model_path, point)
+        return ModelConfig.from_json(output_model.to_json())
 
     def evaluate_model(
-        self, model: OliveModel, data_root: str, metrics: List[Metric], accelerator: AcceleratorSpec
+        self, model_config: ModelConfig, data_root: str, metrics: List[Metric], accelerator: AcceleratorSpec
     ) -> MetricResult:
         """
         Evaluate the model
         """
-        if isinstance(model, CompositeOnnxModel):
+        if model_config.type == "CompositeOnnxModel":
```

Comment:
It might be safer to check with `.lower()` since the keys in the registry is in lower case and the `type` is supposed to be case insensitive. 
Maybe we can also add the `.lower()` step in the type validation to be consistent. 
![image](https://github.com/microsoft/Olive/assets/94929125/34175271-0b2e-40c0-bde3-537a2382bd10)

Reply:
I prefer to keep the original type user specified in ModelConfig. Instead, we could add lower() whenever it need check user inputs. 

Will add lower() when we explicitly check the type.